addons/website_slides/controllers/thread.py

```diff
@@ -0,0 +1,37 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.addons.mail.controllers import thread
+from odoo import _, http
+from odoo.http import request
+from odoo.exceptions import ValidationError
+
+
+class ThreadController(thread.ThreadController):
+
+    @http.route()
+    def mail_message_post(self, thread_model, thread_id, post_data, context=None, **kwargs):
+        previous_post = request.env["mail.message"].search(
+            [
+                ("res_id", "=", thread_id),
+                ("author_id", "=", request.env.user.partner_id.id),
+                ("model", "=", "slide.channel"),
+                ("subtype_id", "=", request.env.ref("mail.mt_comment").id),
+            ]
+        )
+        if thread_model == "slide.channel" and previous_post:
+            raise ValidationError(_("Only a single review can be posted per course."))
```

Comment:
It would be best to check the model before doing the query.

Also the check should only be done if the post_data subtype is comment, as this will prevent from posting as many notes as wanted in current state (don't forget it is now common with backend).